src/date-picker/date-picker.jsx

```diff
@@ -111,7 +117,9 @@ let DatePicker = React.createClass({
     this.setState({
       date: d,
     });
-    this.refs.input.setValue(this.props.formatDate(d));
+    if (!this.refs.input._isControlled()) {
+      this.refs.input.setValue(this.props.formatDate(d));
+    }
```

Comment:
We use this naming convention to indicate that `_isControlled()` is private to the component. Since we're passing all the props to the TextField input anyways, you should be able to determine this based on the props being passed in to date picker.